projects/packages/classic-theme-helper/src/class-featured-content.php

```diff
@@ -464,9 +464,9 @@ public static function hide_featured_term( $terms, $taxonomies, $args ) {
 		 *
 		 * @uses Featured_Content::get_setting()
 		 *
-		 * @param array $terms A list of term objects. This is the return value of get_the_terms().
-		 * @param int   $id The ID field for the post object that terms are associated with.
-		 * @param array $taxonomy An array of taxonomy slugs.
+		 * @param array  $terms A list of term objects. This is the return value of get_the_terms().
```

Comment:
Based on https://developer.wordpress.org/reference/hooks/get_the_terms/, looks like this could be changed to
```suggestion
		 * @param \WP_Term[]|\WP_Error  $terms A list of term objects. This is the return value of get_the_terms().
```
In light of that, possibly we should add some sort of
```php
// It's WP_Error or something invalid from another hook.
if ( ! is_array( $terms ) ) {
    return $terms;
}
```
early on in the function too. Right now we might wind up iterating the properties of a `WP_Error` instead of an array of terms.

Reply:
d0a7d54309